src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -312,49 +312,58 @@ TER Transactor::consumeSeqOrTicket (SLE::pointer const& sleAccount)
     if (seqOrT.isSeq())
     {
         sleAccount->setFieldU32 (sfSequence, seqOrT.value() + 1);
+        return tesSUCCESS;
     }
-    else
+    return ticketDelete (
+        view(), account_, getTicketIndex (account_, seqOrT.value()), j_);
+}
+
+// Remove a single Ticket from the ledger.
+TER Transactor::ticketDelete (ApplyView& view,
+    AccountID const& account, uint256 const& ticketIndex, beast::Journal j)
+{
+    // Delete the Ticket, adjust the account root ticket count, and
+    // reduce the owner count.
+    SLE::pointer const sleTicket = view.peek (keylet::ticket (ticketIndex));
+    if (! sleTicket)
     {
-        // Delete the Ticket, adjust the account root ticket count, and
-        // reduce the owner count.
-        uint256 const ticketIndex =
-            {getTicketIndex (account_, seqOrT.value())};
+        JLOG(j.fatal()) << "Ticket disappeared from ledger.";
+        return tefBAD_LEDGER;
+    }
 
-        SLE::pointer const sleTicket = view().peek (
-            keylet::ticket (ticketIndex));
-        if (! sleTicket)
-        {
-            JLOG(j_.fatal()) << "Ticket disappeared from ledger.";
-            return tefBAD_LEDGER;
-        }
+    std::uint64_t const page {(*sleTicket)[sfOwnerNode]};
+    if (! view.dirRemove (
+        keylet::ownerDir (account), page, ticketIndex, true))
+    {
+        JLOG(j.fatal()) << "Unable to delete Ticket from owner.";
+        return tefBAD_LEDGER;
+    }
 
-        std::uint64_t const page = {(*sleTicket)[sfOwnerNode]};
-        if (! view().dirRemove (
-            keylet::ownerDir (account_), page, ticketIndex, true))
-        {
-            JLOG(j_.fatal()) << "Unable to delete Ticket from owner.";
-            return tefBAD_LEDGER;
-        }
+    // Update the account root's TicketCount.  If the ticket count drops to
+    // zero remove the (optional) field.
+    auto sleAccount = view.peek (keylet::account(account));
```

Comment:
I explored this.  The complication is that `Transactor::ticketDelete()` is also being called from DeleteAccount.cpp.  I put some effort into keeping the number of parameters that DeleteAccount needs to pass minimized, since it also impacts three additional related functions.  So I could make the change you're suggesting.  But in deference to DeleteAccount I'd prefer not too.

Yeah, I acknowledge that `Keylet` computation is expensive.  So if you would like we can revisit this discussion.  I'm to dead set that I'm making the right call.

Reply:
Actually, I realized that the way you're passing in the `ticketIndex`, you've already done the hard work, so creating the `Keylet` is computationally cheap. So this is fine as-is.